### ISSUE TITLE
Upload sourcemaps - can't configure repo url

### ISSUE THREAD

**Issue by weslleyaraujo** [DataDog/datadog-ci#267]
**Describe what happened**

Hi there,

I am having trouble getting the "repo url" working on my RUM application, as seen on the screenshot:

<img width="407" alt="Screen Shot 2021-06-08 at 15 53 20" src="https://user-images.githubusercontent.com/2286385/121199158-feb47b80-c872-11eb-8946-dc7e607e1161.png">

Everything else works fine, only the repo url doesn't seem to be set even though I am sending it via `--repository-url` as described in the docs.

My full command looks like (omitting private information)

```bash
yarn datadog-ci sourcemaps upload ./.next/static \
  --project-path=./src/ \
  --service=my-service \
  --release-version=$(jq -r .version package.json) \
  --minified-path-prefix=https://website-url.com/static \
  --repository-url=https://github.com/company/repo-name/
```

**Steps to reproduce the issue:**

Run the command:

```bash
yarn datadog-ci sourcemaps upload ./.next/static \
  --project-path=./src/ \
  --service=my-service \
  --release-version=$(jq -r .version package.json) \
  --minified-path-prefix=https://website-url.com/static \
  --repository-url=https://github.com/company/repo-name/
```

**Expected behavior:**

The "view repo" on datadog's dashboard should be active

**Actual behavior:**

The "view repo" on datadog's dashboard is not active

**Additional context**
 - Node version: 16.x

--

1. I am running this command from inside docker during my build phase, a simplified version of my `Dockerfile` looks like this:

```docker
FROM node:16

ENV NODE_ENV=production
ENV DATADOG_SITE=datadoghq.eu
ENV DATADOG_API_KEY="<my-api-key>"
RUN apt-get update -y
RUN apt-get install -y jq

# Create app directory
RUN mkdir -p /usr/src/app
WORKDIR /usr/src/app
COPY package.json /usr/src/app/
COPY yarn.lock /usr/src/app/
RUN yarn install

COPY . /usr/src/app
RUN yarn build

RUN yarn datadog-ci sourcemaps upload ./.next/static \
  --project-path=./src/ \
	--service=rekkicom \
	--release-version=$(jq -r .version package.json) \
	--minified-path-prefix=https://website-url.com/static \
	--repository-url=https://github.com/company/repo-name/

EXPOSE 8080
CMD yarn start --port=$PORT

```

2. The repo that I am using is private , not sure if this changes anything

3. I've tried adding `COPY .git /usr/src/app/` assuming the cli needs the `.git` directory there to get some info from it but it didn't work as well

Let me know if I am missing something obvious there, thank you!

**Comment by jybp** [comment]
Hey @weslleyaraujo!

> The repo that I am using is private , not sure if this changes anything
It doesn't change anything, it should have no impact.

> ... assuming the cli needs the .git directory there to get some info from
Yes that is correct. The cli does invoke the git cli in the CWD to get some info.

Make sure the git program is available and can be invoked. Make sure the location you invoke `datadog-ci` from is within the repository.
I'd advise to keep incrementing the `--release-version` if you're re-running the CLI so it doesn't conflict with previous uploads.

If that does not solve the issue, could you please open a support ticket? We would need to gather additional information to investigate.

**Comment by weslleyaraujo** [comment]
Hi @jybp , thanks for the quick answer here!

> Make sure the git program is available and can be invoked. 

Got it, I think that's the root of the issue then since I am running the command inside docker 

> I'd advise to keep incrementing the --release-version

That's happening already, thank you for the tip!

Will close this issue then, if I cant make will reach out via support. Thank you!

**Comment by strowk** [comment]
I also have same issue, but we are using bitbucket, not github and I am sure that we do have git CLI as well as .git folder, because I use it right before calling datadog-ci to get git hash for version like this `build_version="$(cat package.json | jq '.version' -r)-$(git rev-parse --short HEAD)"`. 
Any ideas why it does not work with bitbucket?

**Comment by strowk** [comment]
Hi @jybp  , is there any documentation what are prerequisites for this to work? I want to firstly figure out if I did everything right. For example I think that remote is probably configured to `ssh`, but I am giving https link to `repository-url`. I cannot find any documentation how this feature is supposed to work. Should datadog-ci somehow get access to bitbucket API? I would like to avoid that, do not see why it would be necessary..

**Comment by strowk** [comment]
Ok, after reading sources of this tool, I have one question - 

in this place 
https://github.com/DataDog/datadog-ci/blob/c64a6977bd7b273e60925f439db5c874f41903c4/src/commands/sourcemaps/upload.ts#L149 
I take it that if at least one of sourcemaps have a problem - none of them would be uploaded, correct? 
Seems that in our sources  we have a map looking like this 
`"sources":["../webpack/bootstrap"]` 
and there is no such folder (I take it either this map is broken or it somehow magically works with webpack, no idea). 

And I can see this message 
`No tracked files found for sources contained in ../build/static/js/runtime-main.daf808a6.js.map` 
in logs. Do I read this right that because of one broken source map - none received that "repository info"?

**Comment by jybp** [comment]
👋 

> is there any documentation what are prerequisites for this to work? I want to firstly figure out if I did everything right. For example I think that remote is probably configured to ssh, but I am giving https link to repository-url

There's no constraint on the protocol of the remote or `--repository-url` provided. I don't think we have much more doc apart from that wording in the [readme](https://github.com/DataDog/datadog-ci/tree/master/src/commands/sourcemaps#readme):

```
prevents the command from invoking git in the current working directory and sending repository related data to Datadog (hash, remote URL and the paths within the repository of the sources referenced in the sourcemap).
```

> I take it that if at least one of sourcemaps have a problem - none of them would be uploaded, correct?

That's not correct, a single sourcemap that has any issue should not impact the others. It'll just have an undefined `repositoryPayload`.
If a particular sourcemap file is invalid or has nothing that could match a tracked file in its `"sources":`, no repository data will be associated with that particular sourcemap, but the sourcemap itself will still be uploaded. That is the meaning of the log message you are seeing.
Every other sourcemaps uploaded without that log message showing up would have repository data properly associated with them. 
Unless all of the uploaded sourcemaps have that log message outputted, I don't think you're doing something wrong.
Is the unminification properly working? I'd advise to open a support ticket so we can dig deeper into your issue.

**Comment by strowk** [comment]
> Is the unminification properly working?
Yes, it does work. I have basically same situation as described in this issue descriotion ( first two ticks are green, third is red ), except I do have `git` installed in the container where this runs and I am sure it is not a problem to get versioning information, since I did it right before sending sourcemaps..

I talked this over on live-chat with support, provided all the logs, etc, there is a support ticket, which I guess will take some time to solve. 

But I still think that this feature is very much undocumented and it should be solved somehow either in this project or somewhere on datadog website..

**Comment by jybp** [comment]
@strowk I think I might know what the issue is. We currently only support repository URLs hosted on github.com/gitlab.com/bitbucket.org.
We currently do not create links to the source code if we cannot detect one of those code hosting website.

We do have fallback solutions that are planned to support any kind of URL. (I cannot give you an ETA yet as they have not yet been prioritised)

> But I still think that this feature is very much undocumented 

I agree that's a good point. I'll update the command readme to document explicitly how the feature works and the limitations.

**Comment by strowk** [comment]
> We currently only support repository URLs hosted on github.com/gitlab.com/bitbucket.org.
 
 Yeah, I believe this have to be documented before releasing this feature. Took quite a time trying to configure this in different ways. I hope other features you will do here, would be documented better in the future

**Comment by vitkusv** [comment]
What would help a lot is once the cli is run, it could give a summary of what it did resolve, eg. repo url, etc. and what it failed to resolve - some sort of verbose mode to see the results and troubleshoot quicker.

**Comment by agestasaputra** [comment]
```
Step 10/13 : RUN DATADOG_API_KEY=$DD_API_KEY yarn run datadog-ci sourcemaps upload ./dist/_nuxt --service=$DD_SERVICE --release-version=$BITBUCKET_TAG --minified-path-prefix=/static/js
 ---> Running in 6c0344a5fa28
Starting upload with concurrency 20. 
Will look for sourcemaps in dist/_nuxt
Will match JS files for errors on files starting with /static/js
version: 1.18.18.17-dev service: $DD_SERVICE project path: 
⚠️ No tracked files found for sources contained in dist/_nuxt/03c9e46.js.map
⚠️ No tracked files found for sources contained in dist/_nuxt/056f6b1.js.map
⚠️ No tracked files found for sources contained in dist/_nuxt/080dd4b.js.map
⚠️ No tracked files found for sources contained in dist/_nuxt/0a946a7.js.map
⚠️ No tracked files found for sources contained in dist/_nuxt/0cb9a94.js.map
⚠️ No tracked files found for sources contained in dist/_nuxt/0e6d077.js.map
⚠️ No tracked files found for sources contained in dist/_nuxt/120bc2a.js.map
⚠️ No tracked files found for sources contained in dist/_nuxt/14ae056.js.map
⚠️ No tracked files found for sources contained in dist/_nuxt/1669b3e.js.map
⚠️ No tracked files found for sources contained in dist/_nuxt/16fa232.js.map
⚠️ No tracked files found for sources contained in dist/_nuxt/17d4d15.js.map
⚠️ No tracked files found for sources contained in dist/_nuxt/fb67a44.js.map
⚠️ No tracked files found for sources contained in dist/_nuxt/fbe3b27.js.map
⚠️ No tracked files found for sources contained in dist/_nuxt/fd0e837.js.map
⚠️ No tracked files found for sources contained in dist/_nuxt/fd9a45a.js.map
⚠️ No tracked files found for sources contained in dist/_nuxt/febb52b.js.map
⚠️ No tracked files found for sources contained in dist/_nuxt/febe8bf.js.map
⚠️ No tracked files found for sources contained in dist/_nuxt/ffce349.js.map
Uploading sourcemap dist/_nuxt/01287db.js.map for JS file available at /static/js/01287db.js
Uploading sourcemap dist/_nuxt/0159ae9.js.map for JS file available at /static/js/0159ae9.js
Uploading sourcemap dist/_nuxt/02555f8.js.map for JS file available at /static/js/02555f8.js
Uploading sourcemap dist/_nuxt/02a292d.js.map for JS file available at /static/js/02a292d.js
Uploading sourcemap dist/_nuxt/0345d53.js.map for JS file available at /static/js/0345d53.js
Uploading sourcemap dist/_nuxt/03c9e46.js.map for JS file available at /static/js/03c9e46.js
Uploading sourcemap dist/_nuxt/056f6b1.js.map for JS file available at /static/js/056f6b1.js
Uploading sourcemap dist/_nuxt/05c8546.js.map for JS file available at /static/js/05c8546.js
Uploading sourcemap dist/_nuxt/05e2ea9.js.map for JS file available at /static/js/05e2ea9.js
Uploading sourcemap dist/_nuxt/069c1ee.js.map for JS file available at /static/js/069c1ee.js
Uploading sourcemap dist/_nuxt/080496e.js.map for JS file available at /static/js/080496e.js
Uploading sourcemap dist/_nuxt/080dd4b.js.map for JS file available at /static/js/080dd4b.js
Uploading sourcemap dist/_nuxt/0a6d144.js.map for JS file available at /static/js/0a6d144.js
Uploading sourcemap dist/_nuxt/0a946a7.js.map for JS file available at /static/js/0a946a7.js
Uploading sourcemap dist/_nuxt/0af3298.js.map for JS file available at /static/js/0af3298.js
Uploading sourcemap dist/_nuxt/20446ac.js.map for JS file available at /static/js/20446ac.js
Uploading sourcemap dist/_nuxt/207074a.js.map for JS file available at /static/js/207074a.js
Uploading sourcemap dist/_nuxt/2090759.js.map for JS file available at /static/js/2090759.js
Uploading sourcemap dist/_nuxt/2090fed.js.map for JS file available at /static/js/2090fed.js
Uploading sourcemap dist/_nuxt/21dbfb0.js.map for JS file available at /static/js/21dbfb0.js
Uploading sourcemap dist/_nuxt/21ff48c.js.map for JS file available at /static/js/21ff48c.js
Uploading sourcemap dist/_nuxt/225440f.js.map for JS file available at /static/js/225440f.js
Uploading sourcemap dist/_nuxt/227d093.js.map for JS file available at /static/js/227d093.js
Uploading sourcemap dist/_nuxt/22bf477.js.map for JS file available at /static/js/22bf477.js
Command summary:
✅ Uploaded 441 sourcemaps in 5.24 seconds.
Done in 6.50s.
```

Hey @jybp , i have same issue. I have several question to you:
1. Can you tell me what's the meaning of this error log `No tracked files found for sources contained in dist/_nuxt/ffce349.js.map`? Does't it appear because **there is no source_map file in that path**?
2. Can you tell me what is the purpose of this command `datadog-ci sourcemaps upload ./dist/_nuxt --service=$DD_SERVICE --release-version=$BITBUCKET_TAG --minified-path-prefix=/static/js`? I mean this command isn't for uploading the source-map to static folder in our server, but the command only tell the datadog that where is the source map exist. Does it correct?
3. Do i need upload our source map as manually, like we need a copy our dist/_nuxt folder to our static folder in our server? Does it correct?
